File: Problems_and_Solutions/days-10 to 19/day-12/Java/Solution3A.java
class Solution3A {
    static class Node {
        int data;
        Node next;

        Node(int data) {
            this.data = data;
            this.next = null;
        }
    }

    public static Node deleteNode(Node head, int x) {
        if (head == null) {
            return null;
        }

        if (x == 1) {
            return head.next;
        }

        int count = 1;
        Node prev = null;
        Node curr = head;

        while (curr != null) {
            if (count == x) {
                if (prev != null && prev.next != null) {
                    prev.next = prev.next.next;
                }
                break;
            }
            count++;
            prev = curr;
            curr = curr.next;
        }

        return head;
    }

    public static void printList(Node head) {
        Node curr = head;
        while (curr != null) {
            System.out.print(curr.data);
            if (curr.next != null) System.out.print(" -> ");
            curr = curr.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Node head = new Node(1);
        head.next = new Node(5);
        head.next.next = new Node(2);
        head.next.next.next = new Node(9);

        int x = 2;

        printList(head);
        head = deleteNode(head, x);
        printList(head);
    }
}

File: Problems_and_Solutions/days-10 to 19/day-12/c++/solution 3a.cpp
class Solution {
  public:
    /* Function to delete a node from a linked list */
    Node* deleteNode(Node* head, int x) {
        // cdoe here
        
        Node* trav=head;
        Node* tempNode=head;
        if(x==1)
            head=head->next;
        else
        {
            while(x-1!=1)
            {
                trav=trav->next;
                x--;
            }
            tempNode=trav->next;
            trav->next=trav->next->next;
        }
        delete tempNode;
        return head;
    }
};